### PR TITLE
fix: Regenerate partition stats after stats task for clustering compaction

### PR DESCRIPTION
**Pull request by xiaocai2333**
issue: #33744

### CODE REVIEW

File: internal/datacoord/compaction_task_clustering.go [hunk 1]
```diff
@@ -295,22 +298,89 @@ func (t *clusteringCompactionTask) processMetaSaved() error {
 
 func (t *clusteringCompactionTask) processStats() error {
 	// just the memory step, if it crashes at this step, the state after recovery is CompactionTaskState_statistic.
+	existNonStats := false
+	tmpToResultSegments := make(map[int64][]int64, len(t.GetTmpSegments()))
 	resultSegments := make([]int64, 0, len(t.GetTmpSegments()))
 	for _, segmentID := range t.GetTmpSegments() {
 		to, ok := t.meta.(*meta).GetCompactionTo(segmentID)
 		if !ok {
```
Comment (gland1) [inline internal/datacoord/compaction_task_clustering.go:306]: I saw in some cases ok && to ==nil so I think it should be if !ok or to == nil

File: internal/datacoord/compaction_task_clustering.go [hunk 2]
```diff
@@ -295,22 +298,89 @@ func (t *clusteringCompactionTask) processMetaSaved() error {
 
 func (t *clusteringCompactionTask) processStats() error {
 	// just the memory step, if it crashes at this step, the state after recovery is CompactionTaskState_statistic.
+	existNonStats := false
+	tmpToResultSegments := make(map[int64][]int64, len(t.GetTmpSegments()))
```
Comment (gland1) [inline internal/datacoord/compaction_task_clustering.go:302]: there should be a different behavior depending if stats task is enabled or not

Comment (xiaocai2333) [reply internal/datacoord/compaction_task_clustering.go]: wooo! you catch an important point, I will add it.

**Pull request merged**